src/central_limit_theorem/CentralLimitGraph.jsx

```diff
@@ -218,6 +218,12 @@ export class CentralLimitGraph extends Component {
             MIN_BIN,
             MAX_BIN);
 
+        let activeSampleMeansData = [{
+            x: sampleMeansGraphData[0][0],
+            y: sampleMeansGraphData[0][1],
+            datum: sampleMeansGraphData[0][2],
```

Comment:
To prevent crashes, it's good practice to always check that the array contains the indices that you're using, especially for a nested array like this. I would change this to:

```
        let activeSampleMeansData = [];
        if (sampleMeansGraphData.length > 0 && sampleMeansGraphData[0].length > 2) {
            activeSampleMeansData.push({
                x: sampleMeansGraphData[0][0],
                y: sampleMeansGraphData[0][1],
                datum: sampleMeansGraphData[0][2],
            });
        }
```

Reply:
Got it. I updated it to check length before assigning in the two spots where I process data in this way.

In general, when do you think its necessary to check array length?  In this particular case, the data is generated with the previous function call, i.e. its not like its waiting for a call to an endpoint.  I was kind of on the fence about this, because its almost like catching a bare exception - if the data isn't in those arrays, then I'd almost prefer that it crashes.

Reply:
Yeah, good point.. If you'd rather have a crash than say, an empty graph, then maybe that's fine.

One thing you could do is add an else statement with something like `console.error('Graph data not available!);`, though it seems that eslint discourages `console.error` in the same way it discourages `console.log`. You could also definitely connect this to RavenJS if you want to get these errors in sentry.

I guess because I haven't worked on like, giant JS projects, I've gotten away with just adding a bunch of sanity checks like this, and debugging manually from there. Maybe that's just my style, so if you do want to change it back I'm totally fine with it.

Reply:
No way, I'd say Astro-Sim is a pretty big project.

That's a good point about Raven - I do need to add Raven to this, which I haven't used before.  And to be honest, I prob could use some more sanity checks in my graph components - like with the slider bug I had last week.

Regardless, thanks for the comments!